storage/filesystem.go

```diff
@@ -263,11 +256,18 @@ func (f *Filesystem) WriteFiles(meta metadata.Metadata, files map[string][]byte)
 			return nil
 		}
 
-		// If a fsGroup is defined, Chown all files in the timestamped directory.
+		// Change ownership of the data directory, the timestamp directory and all files within it.
+		paths := []string{
+			f.dataPathForVolumeID(meta.VolumeID),
+			filepath.Join(f.dataPathForVolumeID(meta.VolumeID), tsDirName),
+		}
 		for filename := range files {
-			// Set the uid to -1 which means don't change ownership in Go.
-			if err := os.Chown(filepath.Join(tsDirName, filename), -1, int(*fsGroup)); err != nil {
-				return err
+			paths = append(paths, filepath.Join(f.dataPathForVolumeID(meta.VolumeID), tsDirName, filename))
```

Comment:
Does it matter that we don't set permission on other files in this directory that we may not have explicitly listed in `files`? Do we _really_ want the properties of 'recursive chown' rather than explicitly chown'ing the directory and any other files we are about to write?

... the only concern I can think of with recursive chown, is if someone does something stupid like `ln -s / /path/to/data/dir/something` as we'd then go and chown everything/could be an 'attack vector' to cause the driver ot spend a lot of time doing something useless/harmful.

The only reason I can think for why we *should* recursive chown all, is to avoid issues when cleaning up... though maybe we are better to have Cleanup basically explicitly delete the files we *do* expect to exist, and then attempt to `rmdir` (which will fail if the directory is non-empty). This is probably a better/safer failure mode, so long as we print the "directory not empty" error message in Cleanup :)

Reply:
The latests solution does not use "recursive chown". Instead we just chown the timestamped dir and the files that we created in that dir.

Reply:
Did we do a recursive chown before?

Reply:
No, we did not: https://github.com/cert-manager/csi-lib/pull/65/files